packages/workflows/src/handlers/product/attach-sales-channel-to-products.ts

```diff
@@ -1,32 +1,27 @@
-import { MedusaContainer, ProductTypes } from "@medusajs/types"
-import { EntityManager } from "typeorm"
-import { SalesChannelService } from "../../services"
+import { ProductTypes } from "@medusajs/types"
+import { WorkflowArguments } from "../../helper"
 
 type ProductHandle = string
 type SalesChannelId = string
 
 export async function attachSalesChannelToProducts({
   container,
-  manager,
+  context,
   data,
-}: {
-  container: MedusaContainer
-  manager: EntityManager
-  data: {
-    productsHandleSalesChannelsMap: Map<ProductHandle, SalesChannelId[]>
-    products: ProductTypes.ProductDTO[]
-  }
-}): Promise<void> {
-  const salesChannelService: SalesChannelService = container.resolve(
-    "salesChannelService"
-  )
+}: WorkflowArguments<{
+  productsHandleSalesChannelsMap: Map<ProductHandle, SalesChannelId[]>
```

Comment:
A thought not specific to this handler, but maybe to make these handlers that are associating 2 things, it would a bit more generic if they receive 2 array that must be the same size, and order.
But it is very likely that when the RemoteLink is ready we can have a generic handler to link anything like this.

Reply:
I don't know if I like being dependent on two array that needs to have the same length and order. I think a map is easier to work wit, eventually we can expect an array of object with handle and salesChannelIds, from that we can build a map internally if we want

Reply:
I'm saying this because in this example we have the handle to get the id later, but in another scenario (not specific to product and sales channel) you might not have anything to link 2 entities, and we are doing everything in bulk, the order has to be respected.

Reply:
Yes in that case we will have to compose with arrays, but ideally as long as we have a unique identifier i prefer a map or an array of object from which we can create the map